test/tools/utils.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const { promisify } = require('util');
+
+module.exports = Object.create(null);
+Object.defineProperty(module.exports, '__esModule', { value: true });
+
+/**
+ * callback to pass to array sort
+ */
+module.exports.byStrings = (a, b) => {
+  return `${a}`.localeCompare(`${b}`);
+};
+
+module.exports.sorted = (iterable, how) => {
+  if (typeof how !== 'function') {
+    throw new TypeError('must provide a "how" function to sorted');
+  }
+  const items = Array.from(iterable);
+  items.sort(how);
+  return items;
+};
+
+module.exports.oneMicroTask = promisify(queueMicrotask);
```

Comment:
Nothing wrong with those but this one does communicate meaning of what's required which is the microtask queue be exec-ed. With `sleep(1)` unless I comment it what's to say the `1` isn't meaningful and need to be slept for longer/shorter to get the correct behavior. I don't feel strongly, I could make this use sleep under the hood but keep the name? the main motivation just being to communicate meaning

Reply:
I don't feel very strongly, I can just see where if the implementation changes where the number of promises in the chain changes, then this might not be super reliable and will need to be tweaked anyway, so I wish we had something more general for that await; I almost suggested wrapping our callback function into its own promise so that the callback is what triggered the resolve, and then we could just await it, or doing the whole setup in a function that takes a `done` argument and incorporating the done into the callback, but that seems like too much work